Library/Homebrew/cask/lib/hbc/artifact/uninstall_base.rb

```diff
@@ -226,12 +228,33 @@ def uninstall_delete(*paths)
       def uninstall_trash(*paths)
         return if paths.empty?
 
+        resolved_paths = each_resolved_path(:trash, paths).to_a
+
         ohai "Trashing files:"
-        each_resolved_path(:trash, paths) do |path, resolved_paths|
-          puts path
-          resolved_paths.each { |resolved_path| Utils.gain_permissions(resolved_path, ["-R"], @command) }
-          @command.run!("/usr/bin/xargs", args: ["-0", "--", HOMEBREW_LIBRARY_PATH/"utils/trash.swift"], input: resolved_paths.join("\0"))
-        end
+        puts resolved_paths.map(&:first)
+        trash_paths(*resolved_paths.flat_map(&:last))
+      end
+
+      def trash_paths(*paths)
+        @command.run!("/usr/bin/osascript", args: ["-e", <<-'EOS'.undent, *paths])
+          on run argv
+            repeat with i from 1 to (count argv)
+              set item i of argv to (item i of argv as POSIX file)
+            end repeat
+
+            tell application "Finder"
+              set trashedItems to (move argv to trash)
+              set output to ""
+
+              repeat with i from 1 to (count trashedItems)
+                set item i of trashedItems to POSIX path of (item i of trashedItems as string)
+                set output to output & (item i of trashedItems) & (do shell script "printf \"\\0\"")
+              end repeat
+
+              return output
+            end tell
+          end run
+        EOS
```

Comment:
Also note that in addition to having the trashing sound happen multiple times (which can be funny: it’s so fast it sounds like a paper machine gun), the Finder’s undo would only work for one file at a time (and to a limit). This way, doing undo once will recover everything to their right places.